src/Pact/Gas/Table.hs

```diff
@@ -0,0 +1,173 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE LambdaCase #-}
+module Pact.Gas.Table where
+
+import qualified Data.Map as Map
+
+import Pact.Types.Gas
+import Data.Map (Map)
+--import qualified Data.Map as M
+import Data.Text (Text)
+import qualified Data.Text as T
+
+import Pact.Types.Term
+import Pact.Types.Persistence
+
+-- NB: If pact ends up not having any variadic primitives (currently, I didn't spot any thus far, but also the types don't rule it out)
+-- then the primTableUnreduced here could lose the [Term Ref] function parameter, and simply be Map Text Gas.
+-- Attributing gas costs to Term Ref on its own is problematic in general -- typically we want to account for those costs when the terms
+-- become evaluated, but there may be some cases where the ability to inspect the argument list is helpful, so I'll leave this in for now.
+
+-- The argument lists in the GReduced case may be more useful, but I wasn't able to find instances where GReduced was in present use,
+-- so it's possible we want to simplify that away.
+
+data GasCostConfig = GasCostConfig
+  { _gasCostConfig_primTable :: Map Text ([Term Ref] -> Gas)
+  , _gasCostConfig_selectColumnCost :: Gas -- up-front cost per column in a select operation
+  , _gasCostConfig_readColumnCost :: Gas -- cost per column to read a row
+  , _gasCostConfig_sortFactor :: Gas
+  , _gasCostConfig_moduleCost :: Gas
+  , _gasCostConfig_moduleMemberCost :: Gas
+  , _gasCostConfig_useModuleCost :: Gas
+  , _gasCostConfig_interfaceCost :: Gas
+  , _gasCostConfig_writeCost :: Gas
+  , _gasCostConfig_functionApplicationCost :: Gas
+  }
+
+testGasConfig :: GasCostConfig
+testGasConfig = GasCostConfig
+  { _gasCostConfig_primTable = testGasTable
+  , _gasCostConfig_selectColumnCost = 1
+  , _gasCostConfig_readColumnCost = 1
+  , _gasCostConfig_sortFactor = 10
+  , _gasCostConfig_moduleCost = 1
+  , _gasCostConfig_moduleMemberCost = 1
+  , _gasCostConfig_useModuleCost = 10
+  , _gasCostConfig_interfaceCost = 1
+  , _gasCostConfig_writeCost = 10
+  , _gasCostConfig_functionApplicationCost = 1
+  }
+
+testGasTable :: Map Text ([Term Ref] -> Gas)
+testGasTable =
+  let lookupCost = 10
+  in Map.fromList
+    [ ("describe-module", const 100)
```

Comment:
See above re "test" vs default value population.

Also I think we have to bite the bullet and make an enum for these, as otherwise people will be copypasting the strings with no possible enforcement except runtime. See `Reserved` here for an example: https://github.com/kadena-io/pact/blob/83d3f2b4c32d93d970f6adda825a02d73ea34663/src/Pact/Compile.hs#L83